### PR TITLE
feat(cluster): add VPC feature on the creation flow for AWS

### PR DESCRIPTION
**Pull request by RemiBonnet**
# What does this PR do?

https://qovery.atlassian.net/jira/software/projects/FRT/boards/23?selectedIssue=FRT-1002
https://qovery.atlassian.net/jira/software/projects/FRT/boards/23?selectedIssue=FRT-1010

Capture d’écran 2024-02-26 à 13 45 56

- Missing "resume" tooltip
- Missing cluster feature detail page
Should be done on the next PR 

---

## PR Checklist

- [ ] This PR introduces breaking change(s) and has been labeled as such
- [ ] This PR introduces new store changes
- [ ] I have followed the library pattern i.e `feature`, `ui`, `data`, `utils`
- [x] I made sure the code is type safe (no any)

### CODE REVIEW

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

Qovery can create a Preview Environment for this PR.
To trigger its creation, please post a comment with one of the following command.

| Command | Blueprint environment |
|---|---|
| /qovery []() preview cc1de7de-94e5-40f8-8e45-bc8986d9dfec | [storybook](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/cc1de7de-94e5-40f8-8e45-bc8986d9dfec/services/general) |
 /qovery []() preview 28c47145-c8e7-4b9d-8d9e-c65c95b48425 | [staging](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/28c47145-c8e7-4b9d-8d9e-c65c95b48425/services/general) |
| /qovery []() preview {all\|UUID1,UUID2,...} | To preview multiple environments |

---

This comment has been generated from [Qovery](https://qovery.com) AI :robot:.
Below, a word from its wisdom :
> There is no better time than now, better future does not exist

File: libs/pages/clusters/src/lib/feature/page-clusters-create-feature/step-summary-feature/step-summary-feature.tsx [hunk 1]
```diff
@@ -24,8 +24,18 @@ import { useDocumentTitle } from '@qovery/shared/util-hooks'
 import StepSummary from '../../../ui/page-clusters-create/step-summary/step-summary'
 import { steps, useClusterContainerCreateContext } from '../page-clusters-create-feature'
 
+export function getValueByKey(key: string, data: { [key: string]: string }[] = []): string[] {
+  const result: string[] = []
```
Comment (ctjhoa) [inline libs/pages/clusters/src/lib/feature/page-clusters-create-feature/step-summary-feature/step-summary-feature.tsx:28]: use `reduce`

File: libs/pages/clusters/src/lib/ui/page-clusters-create/step-summary/step-summary.tsx [hunk 2]
```diff
@@ -229,8 +246,64 @@ export function StepSummary(props: StepSummaryProps) {
...
+                {props.featuresData.aws_existing_vpc && (
+                  <>
+                    <li className="mb-2">
+                      VPC ID:{' '}
+                      <strong className="font-medium">{props.featuresData.aws_existing_vpc.aws_vpc_eks_id}</strong>
+                    </li>
+                    <li className="mb-2">
+                      EKS subnets ids:{' '}
+                      <ul className="list-disc ml-4">
+                        {renderSubnets('Zone A:', 'A', props.featuresData.aws_existing_vpc.eks_subnets)}
```
Comment (ctjhoa) [inline libs/pages/clusters/src/lib/ui/page-clusters-create/step-summary/step-summary.tsx:258]: `renderSubnets` should be a component

File: libs/shared/console-shared/src/lib/cluster-settings/ui/card-cluster-feature/card-cluster-feature.tsx [hunk 3]
```diff
@@ -1,28 +1,31 @@
...
 
 export interface CardClusterFeatureProps {
   feature: ClusterFeature
   cloudProvider?: CloudProviderEnum
   disabled?: boolean
-  getValues?: UseFormGetValues<FieldValues>
   setValue?: UseFormSetValue<FieldValues>
+  watch?: UseFormWatch<FieldValues>
   control?: Control<FieldValues>
+  callout?: ReactNode
```
Comment (ctjhoa) [inline libs/shared/console-shared/src/lib/cluster-settings/ui/card-cluster-feature/card-cluster-feature.tsx:13]: can we use `children` instead? It's more future proof

**Review by ctjhoa — CHANGES_REQUESTED** [review]

File: libs/pages/clusters/src/lib/ui/page-clusters-create/step-summary/step-summary.tsx (hunk 2 above)
Comment (RemiBonnet) [reply libs/pages/clusters/src/lib/ui/page-clusters-create/step-summary/step-summary.tsx]: https://github.com/Qovery/console/pull/1192/commits/00b0d54afaaa4086856766dfd72b2bf967da544e

**Review by ctjhoa — APPROVED** [review]

**Pull request merged**